### ISSUE TITLE
Add Loss functions to juliamind.nn.functional.loss

### ISSUE THREAD

**Issue by diksha-shrivastava13** [diksha-shrivastava13/JuliaMind#8]
> Please create a new issue with each of the subtasks and comment "- [ ] #issue_number" to keep the task list updated.

- [ ] dice_loss
- [ ] log_loss
- [ ] fluid_softmax_with_cross_entropy
- [ ] npair_loss
- [ ] square_error_cost
- [ ] edit_distance
- [ ] binary_cross_entropy
- [ ] binary_cross_entropy_with_logits
- [ ] hsigmoid_loss
- [ ] smooth_l1_loss
- [ ] margin_ranking_loss
- [ ] l1_loss
- [ ] nll_loss
- [ ] poisson_nll_loss
- [ ] kl_div
- [ ] mse_loss
- [ ] ctc_loss
- [ ] warpctc
- [ ] rnnt_loss
- [ ] warprnnt
- [ ] margin_cross_entropy
- [ ] softmax_with_cross_entropy
- [ ] cross_entropy
- [ ] sigmoid_focal_loss
- [ ] multi_label_soft_margin_loss
- [ ] hinge_embedding_loss
- [ ] cosine_embedding_loss
- [ ] triplet_margin_with_distance_loss
- [ ] triplet_margin_loss
- [ ] multi_margin_loss
- [ ] soft_margin_loss
- [ ] gaussian_nll_loss
